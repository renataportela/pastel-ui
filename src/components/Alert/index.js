export { default, ALERT_COLORS } from './Alert'