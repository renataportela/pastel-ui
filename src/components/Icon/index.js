export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';