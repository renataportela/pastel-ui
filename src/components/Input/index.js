export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
