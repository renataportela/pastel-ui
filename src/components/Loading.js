import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading