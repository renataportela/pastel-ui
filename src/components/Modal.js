import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;