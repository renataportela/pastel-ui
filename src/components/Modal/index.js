export { default } from './Modal'
export { default as ModalBase } from './ModalBase'