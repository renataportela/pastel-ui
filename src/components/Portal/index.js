export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'