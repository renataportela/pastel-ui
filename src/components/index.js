// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'