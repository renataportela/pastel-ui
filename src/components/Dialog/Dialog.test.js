import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import { Theme, Button, Dialog } from '~/components'

describe('Dialog component', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  afterEach(() => {
    ReactDOM.createPortal.mockClear()
  })

  it('renders a dialog with a Close and OK buttons', () => {
    const tree = renderer.create(
      <Theme>
        <Dialog 
          open
          close={() => {}} 
          title={<p>Hi</p>}
          message={<p>Hello!</p>}
        >        
          <Button label="OK" onClick={() => {}} />
        </Dialog>
      </Theme>
    )
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
})