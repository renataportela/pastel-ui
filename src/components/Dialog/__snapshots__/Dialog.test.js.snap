// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component renders a dialog with a Close and OK buttons 1`] = `
<div
  className="Overlay-sc-1udyqqi-0 evswiq show"
  onClick={[Function]}
>
  <div
    className="Paper__PaperStyle-sc-5lofqf-0 dtOYAq ModalBase__Box-sc-1ykmqu3-0 gHOHhC show"
    onClick={[Function]}
  >
    <h4
      as="span"
      className="Text-sc-7nylk8-0 Heading__HeadingStyle-sc-3bryuh-0 hQqmUM"
      fontSize="4"
      size={null}
      transform={null}
    >
      <p>
        Hi
      </p>
    </h4>
    <p>
      Hello!
    </p>
    <div
      className="sc-AxjAm dBzgwg"
    >
      <button
        className="ButtonStyle-sc-120zjl4-0 fTsNtX"
        color="neutral"
        disabled={false}
        kind="ghost"
        onClick={[Function]}
        size="md"
      >
        Close
        <span
          className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
          onMouseDown={[Function]}
        />
      </button>
      <button
        className="ButtonStyle-sc-120zjl4-0 fJHZAc"
        color="primary"
        disabled={false}
        kind="default"
        onClick={[Function]}
        size="md"
      >
        OK
        <span
          className="Ripple__RippleContainer-sc-1dqnp9a-0 URvnd"
          onMouseDown={[Function]}
        />
      </button>
    </div>
  </div>
</div>
`;
